### ISSUE TITLE
[Bug]: Game crashes when decreasedMonsterRespawnTimer is greater than 2950

### ISSUE THREAD

**Issue by minusnolldag** [MelvorIdle/melvoridle.github.io#3529]
### Describe the bug

Game throws "Error Message: Tried to start timer: Spawn with invalid tick amount: x" when in combat and decreasedMonsterRespawnTimer is greater than 2950 and increaseMonsterRespawnTimer = 0.

### Reproduction Steps

Having all the stuff below unlocked gives the player 1500 (optional 2500). This error is thrown because mods can use the decreasedMonsterRespawnTimer modifier and gives the player more options to decreased MonsterRespawnTimer. But if the player is in game modes HardcoreAdventureSpeedrun, InternalSuffering or HCCOSpeedrun (2000 each), this will cause the game to crash.

The following items need to be equipped:
- Hunters Hat = 200
- Monster scroll = 200

The following pets need to be unlocked:
- Harold = 200

The following agility pillars/obstacles to be unlocked:
- Pillar of combat = 200
- Lava Jump = 300
- Dragon Fight = 300
- Frozen Maze = 100

Optional:
- Township Solar Eclipse = 1000

### Expected behaviour

```javascript
getMonsterSpawnTime() {
	let spawnTime = this.baseSpawnInterval;
	spawnTime -= this.modifiers.decreasedMonsterRespawnTimer;
	spawnTime += this.modifiers.increasedMonsterRespawnTimer;

	// Can be fixed by adding
	if (spawnTime < 50)
		spawnTime = 50;

	return spawnTime;
}

start(time, offsetByTick=false) {
	let ticks = Math.floor(time / TICK_INTERVAL); // TICK_INTERVAL = 50
	if (offsetByTick)
		ticks++;
	if (ticks < 1) // Error happens here
		throw new Error(`Tried to start timer: ${this.type} with invalid tick amount: ${ticks}`);
	this.active = true;
	this._maxTicks = ticks;
	this._ticksLeft = ticks;
    }
````

### Console output

Error due to the following mods:
Archaeology Skill (v1.4)

Error Name: Error
Error Message: Tried to start timer: Spawn with invalid tick amount: -126
Stack Trace:
Error: Tried to start timer: Spawn with invalid tick amount: -126
at Timer.start (https://melvoridle.com/assets/js/built/combatManager.js?6153:265:7)
at CombatManager.loadNextEnemy (https://melvoridle.com/assets/js/built/baseManager.js?6153:46:33)
at CombatManager.loadNextEnemy (https://melvoridle.com/assets/js/built/combatManager.js?6153:176:40)
at CombatManager.checkDeath (https://melvoridle.com/assets/js/built/baseManager.js?6153:28:41)
at Game.tick (https://melvoridle.com/assets/js/built/game.js?6153:178:148)
at Game. (Archaeology Skill:assets/js/patches/game.mjs:12:8)
at typeName. (https://melvoridle.com/assets/js/built/mod.js?6153:839:113)
at Game.processTime (https://melvoridle.com/assets/js/built/game.js?6153:174:6)
at Game.loop (https://melvoridle.com/assets/js/built/game.js?6153:195:50)

Active Action: melvorD:Combat
Current Equipment:
Helmet: melvorTotH:Slayer_Helmet_Mythical
Platebody: melvorTotH:Slayer_Platebody_Mythical
Platelegs: melvorTotH:Vorloran_Protector_Platelegs
Boots: melvorTotH:Vorloran_Protector_Boots
Weapon: melvorTotH:Agile_Wings_Rapier
Shield: melvorTotH:Leviathan_Shield
Amulet: melvorF:Fighter_Amulet
Ring: melvorF:Ring_Of_Wealth
Gloves: melvorTotH:Vorloran_Protector_Gauntlets
Cape: melvorTotH:Superior_Cape_Of_Completion
Passive: melvorF:Chapeau_Noir
Summon1: melvorF:Summoning_Familiar_Cyclops
Summon2: melvorTotH:Summoning_Familiar_Lightning_Spirit

Action Information:
Is In Combat: true
Fight in Progress: false
-- Start of Player Information --
Next Action: Attack
Next Attack: melvorTotH:AgileWingsPlayer
Is Attacking: false
Standard Spell Selected: melvorD:WindStrike
Ancient Spell Selected undefined
Aurora Selected: undefined
Curse Selected: undefined
Active Prayers:melvorTotH:Battleborn, melvorTotH:Valor,
Equipped Food:
Item: melvorD:Whale, Qty: 20700
Item: melvorD:Snape_Grass, Qty: 1059647
Item: melvorD:Whale_Perfect, Qty: 5574
-- End Player Information --
-- Start of Enemy Information --
Enemy: melvorF:AirMonster
Next Action: Nothing
Next Attack: melvorF:WindBlade
Is Attacking: false
Standard Spell Selected: melvorD:WindStrike
Ancient Spell Selected undefined
Aurora Selected: undefined
Curse Selected: undefined
-- End Enemy Information --
Current Area Type: Dungeon
Current Area ID: melvorF:Air_God_Dungeon
Modding Enabled. Loaded mods:
SEMI Core
Combat Simulator Reloaded
Will I Die?
Shop Item Owned Indicator
Completion Log Icons
ETA
QuickShards
Item Uses
Show Skill Modifiers
[Myth] Music
Upgrade All Potions
Archaeology
Treasure Trails - Clue Scrolls
SEMI Drop Chances
Archaeology Skill
Better Summoning Menu
Shining Star
Pokeworld (Generation 1)
QoL - Upgrade X

### Which platforms are you experiencing this bug on?

- [X] Chrome
- [X] Firefox
- [X] Edge
- [X] Safari
- [ ] Steam
- [X] Mobile App (iOS)
- [ ] Mobile App (Android)
- [ ] Mobile App (Android Beta)
- [ ] Other (Please Specify)

### Which version of the game are you experiencing this bug on?

v1.1.2

### Game Subversion

?6154

### Are you using any mods?

Yes

**Comment by Draedon123** [comment]
Can you double check to see if it's still broken? A fix for the mod has been posted already.
![image](https://github.com/MelvorIdle/melvoridle.github.io/assets/118961726/9450dcd0-81c8-42cb-84ab-eb1e32b7710b)

**Comment by minusnolldag** [comment]
Hello Draedon123,
I am the maker of the Archaeology skill mod and I made a temporary fix by using the replace method to address this issue (to make it so other people can play my mod and other mods). Which is why I added the code in this issue that prevents spawnTime from going lower than 50.

**Comment by coolrox95** [comment]
Issue has been fixed as of v1.2,
